src/vs/code/node/cliProcessMain.ts

```diff
@@ -166,7 +166,7 @@ class CliMain extends Disposable {
 
 		// Initialize
 		await Promise.all([
-			stateService.init(),
+			userDataProfilesService.init(),
```

Comment:
This is related to my other comment and debt: you have moved the state service init into the user data profiles service but that is fragile and not good. I would keep the state service init top level and also call user data profiles service init. If you need to join on the state service readiness, either we need a new event from the state service or you pass the init promise of state service into your service.

Reply:
My previous comment will cover this.